tests/foreman/ui/test_contentview.py

```diff
@@ -3967,3 +3967,291 @@ def test_negative_non_readonly_user_actions(module_org, test_name):
+        repo_url=FAKE_0_YUM_REPO
+    )
+    with session:
+        session.settings.update(
+            'name = {}'.format(property_name),
+            param_value
+        )
+        session.contentview.create({
+            'name': cv_name,
+            'solve_dependencies': True,
+        })
+        session.contentview.add_yum_repo(cv_name, repo_name)
+        filter_name = gen_string('alpha')
+        session.contentviewfilter.create(cv_name, {
+            'name': filter_name,
+            'content_type': FILTER_CONTENT_TYPE['package'],
+            'inclusion_type': FILTER_TYPE['exclude'],
+        })
+        session.contentviewfilter.add_package_rule(
+            cv_name, filter_name, package2_name, arch, ('All Versions'))
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == VERSION
+        packages = session.contentview.search_version_package(
+            cv_name, VERSION, 'name = "{}"'.format(package2_name))
+        assert len(packages) == 1
+        assert packages[0]['Name'] == package2_name
+        session.contentviewfilter.add_package_rule(
+            cv_name, filter_name, package1_name, arch, ('All Versions'))
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == 'Version 2.0'
+        for package_names in ['duck', 'cockateel']:
+            packages = session.contentview.search_version_package(
+                cv_name, 'Version 2.0', 'name = "{}"'.format(package_names))
+            assert not packages[0]['Name']
+
+
+@tier2
+def test_positive_conservative_dep_solving_with_multiversion_packages(session, module_org):
+    """Performing solve dependencies on a package with multiple versions that is required
+    by another package.
+
+    :id: f1fb2f2e-6d63-4705-b0cd-6697fed2b6e9
+
+    :setup:
+        1. Turn dependency solving to conservative.
+        2. Create and sync custom repo.
+        3. Create cv and add custom repo.
+        4. Create exclusion filter and exclude a package (walrus) that has multiple versions that
+        is required by another package.
+        5. Publish CV
+        6. Check rpm package list for that version
+        7. Repeat steps 4 - 7 but exclude higher version
+
+    :expectedresults: In step 6, the older version of the package should be filtered out while the
+        newest version is still in the list.  In step 7, the new version is filtered out while the
+        old version is still in the list.
+
+    :CaseImportance: High
+    """
+    property_name = 'dependency_solving_algorithm'
+    param_value = 'conservative'
+    cv_name = gen_string('alpha')
+    repo_name = gen_string('alpha')
+    package_name = 'walrus'
+    arch = 'noarch'
+    create_sync_custom_repo(
+        module_org.id,
+        repo_name=repo_name,
+        repo_url=FAKE_0_YUM_REPO
+    )
+    with session:
+        session.settings.update(
+            'name = {}'.format(property_name),
+            param_value
+        )
```

Comment:
We are updating the setting here, do you think we need to revert the setting for after completion? Just to ensure that we not causing issues with existing tests because of an update in the setting.

Reply:
Good idea.